python/ray/_private/utils.py

```diff
@@ -322,6 +342,129 @@ def get_aws_neuron_core_visible_ids() -> Optional[List[str]]:
+        return None
+    vis_ids = _get_intel_gpu_visible_ids()
+    if vis_ids is None:
+        return None
+    return [str(vis_id) for vis_id in vis_ids]
+
+
+def get_intel_gpu_all_ids() -> Optional[list]:
+    """Get all Intel GPU IDS without ONEAPI_DEVICE_SELECTOR filter,
+        But all Intel GPU still filtered by specific backend and device_type
+    Returns:
+        devices (List[str]): list of strings representing
+        the numeric index (zero-based), with sepcific backend and device_type
+    """
+    selector = os.environ.get("ONEAPI_DEVICE_SELECTOR", None)
+
+    # set ONEAPI_DEVICE_SELECTOR to "backend:device_type"
+    # more info here: https://intel.github.io/llvm-docs/EnvironmentVariables.html
+    backend_type = ray_constants.RAY_ONEAPI_DEVICE_BACKEND_TYPE
+    device_type = ray_constants.RAY_ONEAPI_DEVICE_TYPE
+    os.environ["ONEAPI_DEVICE_SELECTOR"] = backend_type + ":" + device_type
+
+    dev_ids = _get_intel_gpu_visible_ids()
+
+    # set ONEAPI_DEVICE_SELECTOR value back
+    if selector is not None:
+        os.environ["ONEAPI_DEVICE_SELECTOR"] = selector
+
+    return dev_ids
+
+
+def get_intel_gpu_name() -> str:
+    """Get the name of first Intel GPU.
+        Example:
+            name: 'Intel(R) Data Center GPU Max 1550'
+            return name: 'Max-1550'
+    Returns:
+        The name of Intel GPU type.
+    """
+    try:
+        import dpctl
+    except ImportError:
+        dpctl = None
+    if dpctl is None:
+        return ""
+    backend = ray_constants.RAY_ONEAPI_DEVICE_BACKEND_TYPE
+    device_type = ray_constants.RAY_ONEAPI_DEVICE_TYPE
+    devices = dpctl.get_devices(backend=backend, device_type=device_type)
+    if len(devices) > 0:
+        name = devices[0].name
+        return "-".join(name.split(" ")[-2:])
+    return ""
+
+
+def _get_intel_gpu_visible_ids() -> Optional[list]:
+    """Get oneAPI device (GPU) IDs by calling `dpctl` api
+        device with specific backend and device_type
+    Returns:
+        devices IDs (List[str]): return the list of string representing
+        the relative IDs filtered by ONEAPI_DEVICE_SELECTOR, with specificed
+        backend and device_type. The ID is the logic ID.
+        returned visible IDs start with index 0
+    Example:
+        if ONEAPI_DEVICE_SELECTOR="level_zero:2,3,4"
+        the device IDs enumerated will be [0,1,2]
+        same with CUDA_VISIBLE_DEVICES
+    """
+    try:
+        import dpctl
+    except ImportError:
+        dpctl = None
+    if dpctl is None:
+        # Maybe should raise a warning with message:
+        # "Import dpctl error, maybe dpctl not installed."
+        return None
+    selector = os.environ.get("ONEAPI_DEVICE_SELECTOR", None)
+
+    backend = ray_constants.RAY_ONEAPI_DEVICE_BACKEND_TYPE
+    device_type = ray_constants.RAY_ONEAPI_DEVICE_TYPE
+    dev_ids = []
+    # return dev_ids
+    for dev in dpctl.get_devices(backend=backend, device_type=device_type):
```

Comment:
can you please add a comment that the dpctl.get_devices will respect the ONEAPI_DEVICE_SELECTOR environment variable?